### PR TITLE
Allow options to be passed to `get()`

### PR DESCRIPTION
**Pull request by jacobmbr**
This is needed for SFTP downloads that hit this issue: https://github.com/mscdex/ssh2-streams/issues/43

### CODE REVIEW

File: lib/promiseSftp.coffee [hunk 1]
```diff
@@ -295,12 +295,11 @@ class PromiseSftp
...
           sticky: null   # TODO
-    
-    @get = (sourcePath) -> Promise.try () ->
+
+    @get = (sourcePath, options = {}) -> Promise.try () ->
       if restartOffset != null
-        options =
-          start: restartOffset
-          flags: 'r+'
+        options.start = restartOffset
```
Comment (nmccready) [inline lib/promiseSftp.coffee]: Thanks for the PR.
Should it not be?

```
options.start = options.start || restartOffset
options.flags = options.flags || 'r+'
```

Comment (jacobmbr) [reply lib/promiseSftp.coffee]: Agreed for `options.start`. 
Is there a use case for force overriding `r+` for a `get()`?

Comment (nmccready) [reply lib/promiseSftp.coffee]: Just figured if you want to allow full blow options; then allow it. Then people can experiment.

**Comment by zacronos** [comment]
I'm not clear on how it is related to the ssh2 issue you linked, but regardless it's a fine contribution.  Thanks!

**Pull request merged**